src/sdk/namespace_gcp.js

```diff
@@ -7,9 +7,11 @@ const util = require('util');
 const P = require('../util/promise');
 const stream_utils = require('../util/stream_utils');
 const dbg = require('../util/debug_module')(__filename);
+const s3_utils = require('../endpoint/s3/s3_utils');
 const S3Error = require('../endpoint/s3/s3_errors').S3Error;
 // we use this wrapper to set a custom user agent
 const GoogleCloudStorage = require('../util/google_storage_wrap');
+const aws_sdk = require('aws-sdk');
```

Comment:
Why do you use the old SDK (v2 instead of v3) with the new code?
```suggestion
const { S3 } = require('@aws-sdk/client-s3');
```

If you have a good reason for using AWS SDK v2 please keep the same name (easier to search and aligned with all our files):
  ```suggestion
const AWS = require('aws-sdk');
```

Reply:
In my opinion, it is better to write new code with the new AWS SDK (or at least try...).
In this PR you're using an s3 client with basic properties, and specific functions (`createMultipartUpload`, `uploadPart`, `completeMultipartUpload`, `abortMultipartUpload`).
This code would be tested so it will be easier to handle the bugs (shorter cycle).

@liranmauda @dannyzaken What do you think?

Reply:
Need to consider this is done on top of GCP. We know there is compatibility with the previous SDK, and there might be with the new versions as well, but we do need to keep it in mind